### ISSUE TITLE
Add support for Alpine based docker images in OpenTelemetry Operator.

### ISSUE THREAD

**Issue by andretone** [open-telemetry/opentelemetry-dotnet-instrumentation#2658]
# Bug Report

## Symptom

**Describe the bug**

Inside the pod, autoinstrumentation logs `Unable to load shared library 'OpenTelemetry.AutoInstrumentation.Native'`

**Expected behavior**

Autoinstrumentation does not log errors.

**Runtime environment (please complete the following information):**

- OpenTelemetry Instrumentation via Opentelemetry operator: opentelemetry-operator-0.30.0   app.v.: 0.77.0
- OpenTelemetry Automatic Instrumentation version: autoinstrumentation-dotnet:0.7.0
- OS: alpine3.17-amd64
- .NET version: 
  - Host Version: 6.0.16, Architecture: x64, Commit: 1e620a42e7
   .NET runtimes installed: Microsoft.AspNetCore.App 6.0.16 [/usr/share/dotnet/shared/Microsoft.AspNetCore.App] Microsoft.NETCore.App 6.0.16 [/usr/share/dotnet/shared/Microsoft.NETCore.App]

**Additional context**

## Reproduce

 1. use a dotnet instrumentation configuration like this (opentelemetry operator):
     ```
     dotnet:
      env:
      - name: OTEL_DOTNET_AUTO_LOG_DIRECTORY
        value: /tmp
      - name: OTEL_METRICS_EXPORTER
        value: none
      - name: OTEL_EXPORTER_OTLP_ENDPOINT
        value: http://otel-collector-collector.<instrumentationnamespace>:4318
      image: enidgt.azurecr.io/endorsed/autoinstrumentation-dotnet:0.7.0
     ```
2. Create a dotnet pod application with the annotation `instrumentation.opentelemetry.io/inject-dotnet: platform-molo-cli/otel-instrumentation`
3. Observe logs on `/tmp`
    ```
    [Error] EventSource=OpenTelemetry-AutoInstrumentation, Message=Unable to load shared library 'OpenTelemetry.AutoInstrumentation.Native' or one of its dependencies. In order to help diagnose loading problems, consider setting the LD_DEBUG environment variable: Error loading shared library libOpenTelemetry.AutoInstrumentation.Native: No such file or directory
    Exception: Unable to load shared library 'OpenTelemetry.AutoInstrumentation.Native' or one of its dependencies. In order to help diagnose loading problems, consider setting the LD_DEBUG environment variable: Error loading shared library libOpenTelemetry.AutoInstrumentation.Native: No such file or directory
    System.DllNotFoundException: Unable to load shared library 'OpenTelemetry.AutoInstrumentation.Native' or one of its dependencies. In order to help diagnose loading problems, consider setting the LD_DEBUG environment variable: Error loading shared library libOpenTelemetry.AutoInstrumentation.Native: No such file or directory
       at OpenTelemetry.AutoInstrumentation.NativeMethods.NonWindows.InitializeProfiler(String id, NativeCallTargetDefinition[] methodArrays, Int32 size)
       at OpenTelemetry.AutoInstrumentation.NativeMethods.InitializeProfiler(String id, NativeCallTargetDefinition[] methodArrays) in /project/src/OpenTelemetry.AutoInstrumentation/NativeMethods.cs:line 48
       at OpenTelemetry.AutoInstrumentation.Instrumentation.Initialize() in /project/src/OpenTelemetry.AutoInstrumentation/Instrumentation.cs:line 202
```

**Comment by Kielek** [comment]
For now, OpenTelemetry operator is not supporting alpine based docker images (using musl as a dependency instead of glibc).

It is a feature request for OpenTelemetry Operator.

**Comment by Kielek** [comment]
It is available since 0.85.0  and fixed in 0.86.0 release of operator: https://github.com/open-telemetry/opentelemetry-operator/blob/main/CHANGELOG.md#0860. Closing.
